packages/react/src/file-upload/FileUpload.tsx

```diff
@@ -267,6 +267,17 @@ export const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
 			? `${fallbackId}-accepted-files-desc`
 			: '';
 
+		const buttonLabel = `Select ${filesPlural}`;
```

Comment:
Might be a good opportunity to update the variable name `filesPlural` since it can have the value of `'file'`. Perhaps `fileOrFiles`?

Reply:
Yeah, this sounds better